NLP/main/extract_display_quotes.py

```diff
@@ -0,0 +1,531 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Jul  4 14:55:46 2022
+
+@author: sjuf9909
+"""
+
+# import required packages
+import os
+import io
+import sys
+import codecs
+import utils
+import logging
+import traceback
+from collections import Counter
+
+# matplotlib: visualization tool
+from matplotlib import pyplot as plt
+
+# pandas: tools for data processing
+import pandas as pd
+
+# spaCy and NLTK: natural language processing tools for working with language/text data
+import spacy
+from spacy import displacy
+from spacy.tokens import Span
+import nltk
+nltk.download('punkt')
+from nltk import Tree
+from nltk.tokenize import sent_tokenize
+
+# ipywidgets: tools for interactive browser controls in Jupyter notebooks
+import ipywidgets as widgets
+from ipywidgets import Button, Layout
+from IPython.display import display, Markdown, clear_output
+
+# import the quote extractor tool
+# Source: https://github.com/sfu-discourse-lab/GenderGapTracker
+from quote_extractor import extract_quotes, get_rawtext_files
+from config import config
+
+
+class QuotationTool():
+    '''
+    Interactive tool for extracting and displaying quotes in a text
+    '''
+    
+    def __init__(self):
+        '''
+        Initiate the QuotationTool
+        '''
+        # initiate the app_logger
+        self.app_logger = utils.create_logger('quote_extractor', log_dir='logs', 
+                                         logger_level=logging.INFO,
+                                         file_log_level=logging.INFO)
+        
+        # download spaCy's en_core_web_lg, the pre-trained English language tool from spaCy
+        print('Loading spaCy language model...')
+        self.nlp = spacy.load('en_core_web_lg')
+        print('Finished loading.')
```

Comment:
More things can be initiated here? Such as a default configuration including inc_ent and corresponding span visual template. Not sure if empty text_df and quotes_df should also be defined and validate as part of init.

Reply:
empty text_df and quotes_df added to initiation. for inc_ent, I think it flows better when it's added when we get the quotes. but we can initiate at the start as well if u want